dashboard/src/components/OperatorInstance/OperatorInstance.tsx

```diff
@@ -21,19 +23,41 @@ export interface IOperatorInstanceProps {
     crdName: string,
     resourceName: string,
   ) => Promise<void>;
+  deleteResource: (namespace: string, crdName: string, resource: IResource) => Promise<boolean>;
+  push: (location: string) => RouterAction;
   error?: Error;
   resource?: IResource;
   csv?: IClusterServiceVersion;
 }
 
-class OperatorInstance extends React.Component<IOperatorInstanceProps> {
+export interface IOperatorInstanceState {
+  modalIsOpen: boolean;
+  crd?: IClusterServiceVersionCRD;
+}
+
+class OperatorInstance extends React.Component<IOperatorInstanceProps, IOperatorInstanceState> {
+  public state: IOperatorInstanceState = {
+    modalIsOpen: false,
+  };
+
   public componentDidMount() {
     const { csvName, crdName, instanceName, namespace, getResource } = this.props;
     getResource(namespace, csvName, crdName, instanceName);
   }
 
+  public componentDidUpdate(prevProps: IOperatorInstanceProps) {
+    const { resource, csv } = this.props;
+    if (csv !== prevProps.csv || resource !== prevProps.resource) {
```

Comment:
What happens if people change the namespace? I assume the controller tries to get the resource and we'll 404 appropriately so it's not relevant here?

Reply:
good point, I didn't test the namespace change here (TBH I am just testing the happy path for the moment until I have something working end-to-end)